static/js/services/alertify.js

```diff
@@ -89,7 +89,42 @@ define(["angular"], function(angular) {
 			});
 		};
 
+		var registeredCustomDialog = [];
 		var dialog = {
+			/*
+			* @description
+			* Create a custom dialog.
+			*
+			* To overwrite an existing custom dialog simply use the same id
+			*
+			* @param {Object} config Preferences for the dialog with the following properties:
+			* <ul>
+			*   <li>**baseType**: {string} existing dialog type to use as initial values and from which a template will be used.</li>
+			*   <li>**type**: {string} The ID of the custom dialog. The template name which is saved in $templateCache. If the the type is 'notify' the templateUrl must be '/dialogs/notify.html' </li>
+			*   <li>**template**: {string} A custom template to use for the dialog when no baseType is used.</li>
+			*   <li>**title**: {string} the title which the modal dialog should display when no template is provided.</li>
+			*   <li>**message**: {string} the message which the modal dialog should display when no template is provided.</li>
+			*   <li>**ok_cb**: {Function} the callback function to be called on success.</li>
+			*   <li>**err_cb**: {Function} the callback function to be called on error.</li>
+			* </ul>
+			*/
+			registerCustom: function(config) {
+				if (!config ||
+					config && !config.type ||
+					config && !config.template && !config.baseType) {
+					throw Error("Custom template not configured correctly.");
+				}
+				var templateUrl = '/dialogs/' + config.type + '.html';
+				if (config.baseType) {
+					$templateCache.put(templateUrl, $templateCache.get('/dialogs/' + config.baseType + '.html'));
+					if (!config.title) {
+						config.title = api.defaultMessages[config.baseType];
```

Comment:
This changes the mapping provided as parameter. It should not be changed. A clone of the parameter should be stored and that one changed.